src/main/java/com/denizenscript/denizencore/scripts/commands/core/ImageCommand.java

```diff
@@ -0,0 +1,182 @@
+    //
+    // All uses of the image command must include the "id:" argument. This is any arbitrary name, as plaintext or from a tag,
+    // to uniquely and globally identify the image object in memory. This ID can only be used by one image object at a time.
+    // IDs are stored when "load" is used, and only removed when "unload" is used.
+    //
+    // @Tags
+    // None
+    //
+    // @Usage
+    // Use to load an image from a file into memory under the id "image".
+    // - ~image id:image load path:data/image.png
+    //
+    // @Usage
+    // Use to load an image object from a definition, draw on it, then store it back into the definition.
+    // - define image <image[width=50;height=50;background=blue]>
+    // - image id:to_edit load image:<[image]>
+    // - draw id:to_edit oval x:0 y:0 width:50 height:50 color:green
+    // - define image <image[to_edit]>
+    //
+    // @Usage
+    // Use to save an image into file and unload it.
+    // - ~image id:image save path:data/image.png
+    // - image id:image unload
+    // -->
+
+    public static final Map<String, ImageTag> loadedImages = new HashMap<>();
+
+    public ImageCommand() {
+        setName("image");
+        setSyntax("image [id:<id>] [load [image:<image>/path:<path>]]/[save [path:<path>] [format:<format>]]/[unload]");
+        setRequiredArguments(2, 4);
+        autoCompile();
+    }
+
+    public static ImageTag getImageFrom(String id) {
+        ImageTag image = loadedImages.get(CoreUtilities.toLowerCase(id));
+        if (image == null) {
+            throw new InvalidArgumentsRuntimeException("No loaded image with id '" + id + "'.");
+        }
+        return image;
+    }
+
+    public enum Operation {LOAD, SAVE, UNLOAD}
+
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("operation") Operation operation,
+                                   @ArgName("id") @ArgPrefixed String id,
+                                   @ArgName("path") @ArgPrefixed @ArgDefaultNull String path,
+                                   @ArgName("format") @ArgPrefixed @ArgDefaultText("png") String format,
+                                   @ArgName("image") @ArgPrefixed @ArgDefaultNull ImageTag toLoad) {
+        String idLower = CoreUtilities.toLowerCase(id);
+        switch (operation) {
+            case LOAD -> {
+                if (toLoad == null && path == null) {
+                    throw new InvalidArgumentsRuntimeException("Must specify a path or image to load.");
+                }
+                if (loadedImages.containsKey(idLower)) {
+                    Debug.echoError(scriptEntry, "Id '" + id + "' is already in use.");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                if (toLoad != null) {
+                    ImageTag toLoadCopy = toLoad.duplicate();
```

Comment:
This block needs an update for images-identified-by-id change

Reply:
Mended